src/type.h

```diff
@@ -55,18 +55,30 @@ class Type {
     I32U = 7,  // Not actually specified, but used internally with load/store
   };
 
+  enum class Nullable : bool {
+    No = false,
+    Yes = true,
+  };
+
   Type() = default;  // Provided so Type can be member of a union.
-  Type(int32_t code) : enum_(static_cast<Enum>(code)) {}
-  Type(Enum e) : enum_(e) {}
+  Type(int32_t code, Nullable nullable = Nullable::Yes)
+      : enum_(static_cast<Enum>(code)), nullable(nullable) {}
+  Type(Enum e) : enum_(e), nullable(Nullable::Yes) {}
   operator Enum() const { return enum_; }
 
   bool IsRef() const {
+    return enum_ == Type::ExternRef || enum_ == Type::FuncRef || IsIndex();
```

Comment:
Why does the definition of `IsRef` need to change here?

Reply:
According to the spec (https://github.com/WebAssembly/function-references/blob/master/proposals/function-references/Overview.md#reference-types) we have new definition of the ref type - now it is a heaptype + nullability flag, in turn heaptype ::= <typeidx> | func | extern.